cfg/pre-commit-config.yaml

```
@@ -7,7 +8,53 @@ repos:
     entry: './src/sh/ensure-shebang.sh'
```

How about we start with a hook that

1. Asserts poetry is installed (and prints an error message if not)
2. Poetry installs the package at `src/python/hooks/pyproject.toml`

This way when running locally the user will get feedback that they need to get poetry, and when run on CI step 1 will just silently return

---

This is done now, I believe I covered all of the bases and error messages. It's a bit more convoluted than I thought it would be, so hopefully I didn't miss anything, but I focused on making sure things work in CI rather than locally, because the developer can always figure out how to fix things locally (or we can add it to a README or something)